contracts/DelFiPrice.sol

```diff
@@ -27,17 +43,47 @@ contract DelFiPrice is OpenOracleView {
     /// @notice The mapping of medianized prices per symbol
     mapping(string => uint64) public prices;
 
+    /// @notice The structure that stores addresses of CToken contracts
+    CTokens public tokens;
+
+    /// @notice The binary representation for 'ETH' symbol , used for string comparison
+    bytes32 constant symbolEth = keccak256(abi.encodePacked("ETH"));
+
+    /// @notice The binary representation for 'USDC' symbol, used for string comparison
+    bytes32 constant symbolUsdc = keccak256(abi.encodePacked("USDC"));
+
+    /// @notice The binary representation for 'DAI' symbol, used for string comparison
+    bytes32 constant symbolDai = keccak256(abi.encodePacked("DAI"));
+
+    /// @notice The binary representation for 'REP' symbol, used for string comparison
+    bytes32 constant symbolRep = keccak256(abi.encodePacked("REP"));
+
+    /// @notice The binary representation for 'BTC' symbol, used for string comparison
+    bytes32 constant symbolWbtc = keccak256(abi.encodePacked("BTC"));
+
+    /// @notice The binary representation for 'BAT' symbol, used for string comparison
+    bytes32 constant symbolBat = keccak256(abi.encodePacked("BAT"));
+
+    /// @notice The binary representation for 'ZRX' symbol, used for string comparison
+    bytes32 constant symbolZrx = keccak256(abi.encodePacked("ZRX"));
+
     /**
      * @param data_ Address of the Oracle Data contract
      * @param sources_ The reporter addresses whose prices will be used to calculate the median
      * @param anchor_ The reporter address whose prices checked against the median for safety
      * @param anchorToleranceMantissa_ The tolerance allowed between the anchor and median. A tolerance of 10e16 means a new median that is 10% off from the anchor will still be saved
+     * @param tokens_ The CTokens struct that contains addresses for CToken contracts
      */
-    constructor(OpenOraclePriceData data_, address[] memory sources_, address anchor_, uint anchorToleranceMantissa_) public OpenOracleView(data_, sources_) {
+    constructor(OpenOraclePriceData data_, 
+                address[] memory sources_,
+                address anchor_,
+                uint anchorToleranceMantissa_,
+                CTokens memory tokens_) public OpenOracleView(data_, sources_) {
         anchor = anchor_;
         require(anchorToleranceMantissa_ < 100e16, "Anchor Tolerance is too high");
         upperBoundAnchorRatio = 100e16 + anchorToleranceMantissa_;
         lowerBoundAnchorRatio = 100e16 - anchorToleranceMantissa_;
+        tokens = tokens_;
```

Comment:
we probably want to destructure the tokens into immutable state variables to save all that gas in the lookup function. Would we awesome if immutable structs were a thing

Reply:
cool, so I will change that for this variant with structure. And then we can look at how 2 dimensional array variant looks like as well.